skrub/_drop_uninformative.py

```diff
@@ -0,0 +1,137 @@
+import numbers
+
+from sklearn.utils.validation import check_is_fitted
+
+from . import _dataframe as sbd
+from ._on_each_column import SingleColumnTransformer
+
+__all__ = ["DropUninformative"]
+
+
+class DropUninformative(SingleColumnTransformer):
+    """Drop column if it is found to be uninformative according to various criteria.
+
+    A column is considered to be "uninformative" if one or more of the following
+    issues are found:
+    - The fraction of missing values is larger than a certain fraction (by default,
+    all values must be null for the column to be dropped).
+    - The column includes only one unique value (the column is constant). Missing
+    values are considered a separate value.
+    - The number of unique values in the column is equal to the length of the column.
+
+    Parameters
+    ----------
+    constant_column: bool, default=True
+        If True, drop the column if it contains only one unique value. Missing values
+        count as one additional distinct value.
+    column_is_id: bool, default=False
+        If True, drop the column if all values are distinct. Missing values count as
+        one additional distinct value.
+    threshold: float or None, default=1.0
```

Comment:
now that there are several criteria I think it should be called something like `null_fraction_threshold`